internal/store/fleet.go

```diff
@@ -111,28 +133,77 @@ func (s *FleetStore) CreateOrUpdateFleet(ctx context.Context, orgId uuid.UUID, r
 	fleet := model.NewFleetFromApiResource(resource)
 	fleet.OrgID = orgId
 
+	// don't allow the user to set the generation
+	fleet.Generation = nil
+	if fleet.Spec != nil && fleet.Spec.Data.Template.Metadata != nil {
+		fleet.Spec.Data.Template.Metadata.Generation = nil
+	}
+
 	// don't overwrite status
 	fleet.Status = nil
 
 	created := false
-	findFleet := model.Fleet{
-		Resource: model.Resource{OrgID: orgId, Name: *resource.Metadata.Name},
-	}
-	result := s.db.First(&findFleet)
-	if result.Error != nil {
-		if result.Error == gorm.ErrRecordNotFound {
-			created = true
+
+	err := s.db.Transaction(func(tx *gorm.DB) error {
+		existingRecord := model.Fleet{Resource: model.Resource{OrgID: fleet.OrgID, Name: fleet.Name}}
+		result := tx.First(&existingRecord)
+		if result.Error != nil {
+			if !errors.Is(result.Error, gorm.ErrRecordNotFound) {
+				return result.Error
+			} else {
```

Comment:
nit: the if statement above returns do we need the else statement? Also in general nesting of if conditions should be minimized to improve readability and reduce complexity.

```golang
if result.Error != nil && !errors.Is(result.Error, gorm.ErrRecordNotFound) {
 // something specific
 }
 
if result.Error != nil {
   // general error case
  }

// happy path
 
 ```

Reply:
True

Reply:
Fixed